src/libs/actions/Report.js

```diff
@@ -323,6 +351,82 @@ function fetchChatReportsByIDs(chatList) {
         });
 }
 
+/**
+ * Update stored IOU object for given IOU and chat report ID
+ *
+ * @param {Number} iouReportID
+ * @param {Number} chatReportID
+ */
+function updateIOUReportData(iouReportID, chatReportID) {
```

Comment:
There is already a function named `fetchIOUReportData` which fetches the data but doesn't update stored data. It is used by `fetchChatReportsByIDs` function to update all chat and IOU reports in batch.

Reply:
Ok, but why do we need two separate methods that both call the same APIs? Looking at that method it calls both `GetIOUReport` and `Get: reportStuff`. But it should be calling `fetchIOUReport()` and `fetchIOUReportID()` if we also have methods that only do that.

Reply:
Now we have 4 functions related to iou reports:
- `fetchIOUReport` takes `iouReportID`, `chatReportID` and calls `Get: reportStuff` and returns `iouReportObject`.
- `fetchIOUReportID` takes `debtorEmail` and calls `GetIOUReport` and returns `iouReportID`.

- `fetchIOUReportDataForChatReport` uses `fetchIOUReportID` and `fetchIOUReport` to return `iouReportObject` for given chat report.
- `updateIOUReportData` takes `iouReportID`, `chatReportID` and  calls `fetchIOUReport` to get `iouReportObject` and updates Onyx data.

Reply:
I'm confused it looks like we got rid of `fetchIOUReportDataForChatReport`?

Reply:
The names need to be fixed so they make sense and are easy to understand.

Reply:
I just submitted the commit with these 4 functions. Which one is not clear and should be renamed?

Reply:
`fetchIOUReportDataForChatReport` - doesn't sound very different from `fetchIOUReport()` because all of them are "for chat reports". The method is also only used once so I think there's no reason to create it. For now, could we please just get rid of this method and move the logic here instead to avoid confusion:

https://github.com/Expensify/Expensify.cash/blob/103e1d1199032986013995515dede56c02a08f31/src/libs/actions/Report.js#L312

`updateIOUReportData` is inaccurate because we are _fetching_ data and setting it not updating it. It's not really conventional (for this codebase) to have a method `fetchIOUReport()` that fetches, but doesn't set any data (all of the other actions are fetching and setting data in Onyx). And then have another method `updateIOUReportData()` that both fetches and sets the data.

So, my only suggestion is to remove the calls to `fetchIOUReport()` inside `updateIOUReportData()` and call it before each usage then rename `updateIOUReportData()` to `setLocalIOUReportData()` so it is clear that it will only save the data to Onyx.

Reply:
I updated function names as you recommended. Can you check again?